tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionParserSuite.scala

```diff
@@ -21,6 +21,24 @@ import org.scalatest.FunSuite
 
 class RegularExpressionParserSuite extends FunSuite {
 
+  test("detect regexp strings") {
+    // Based on https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html
+    val strings: Seq[String] = Seq("\\", "\u0000", "\\x00",
+      "\f", "\\a", "\\e", "\\cx", "[abc]", "^", "[a-z&&[def]]", ".", "*", "\\d", "\\D",
+      "\\h", "\\H", "\\s", "\\S", "\\v", "\\V", "\\w", "\\w", "\\p", "$", "\\b", "\\B",
+      "\\A", "\\G", "\\Z", "\\z", "\\R", "?", "|", "(abc)", "a{1,}", "\\k", "\\Q", "\\E")
+    for (string <- strings) {
+      assert(RegexParser.isRegExpString(string))
+    }
+  }
+
+  test("detect non-regexp strings") {
+    val strings = Seq("\\.", "A", ",", "\t")
```

Comment:
We also need to treat `":"` as non-regex.
```suggestion
    val strings = Seq("\\.", "A", ",", "\t", ":")
```